torch/csrc/jit/import_method.cpp

```diff
@@ -0,0 +1,113 @@
+#include "torch/csrc/jit/import_method.h"
+#include "torch/csrc/jit/script/parser.h"
+
+namespace torch { namespace jit {
+
+
+// this is a much simpler accessor that only handles modules, parameters, and
+// and methods
+struct ModuleAccessorValue : public script::SugaredValue {
+  ModuleAccessorValue(std::shared_ptr<script::Module> module)
+  : module(std::move(module)) {}
+  std::string kind() const override {
+    return "module";
+  }
+  // select an attribute on it, e.g. `this.field`
+  std::shared_ptr<SugaredValue> attr(SourceRange loc, script::Method & m, const std::string& field) override {
+    if(script::NamedModule* v = module->find_module(field)) {
+      return std::make_shared<ModuleAccessorValue>(v->module);
+    } else if(script::NamedParameter* v = module->find_parameter(field)) {
+      return std::make_shared<script::SimpleValue>(m.get_or_add_parameter(v->slot()));
+    } else if(script::Method* m = module->find_method(field)) {
+      return std::make_shared<script::MethodValue>(module, *m);
+    }
+    return script::SugaredValue::attr(loc, m, field);
+  }
+private:
+  std::shared_ptr<script::Module> module;
+};
+
+// this is a much simpler accessor that only handles modules, parameters, and
```

Comment:
Can we have a more proper comment on the ConstantTableValue, it is not quite intuitive what is it for and how it's being used.